route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -84,6 +84,7 @@ namespace route_following_plugin
             ROS_WARN_STREAM("Cannot find current lanelet in shortest path!");
             return true;
         }
+        ros::Time current_time=ros::Time::now();
```

Comment:
One last item, can't recall if I've mentioned this prior: I'd recommend at the start of the function checking to see if the service request has any maneuvers in it already. If it doesn't, everything here looks good. If it does, you should use the end time and distance of the last maneuver in that request as the `current_time` and `current_loc` for this planning, with the same target of 15s total plan length. Essentially just add on to the end of the plan if there's already maneuvers there.